src/connection.h

```diff
@@ -114,14 +114,15 @@ typedef struct ConnectionType {
 struct connection {
     ConnectionType *type;
     ConnectionState state;
+    int last_errno;
+    int fd;
```

Comment:
Were these re-ordered for improving memory locality?

Reply:
This change reduces 8 bytes on a 64-bits platform.

Reply:
I can't see how this reduces 8 bytes. Maybe it depends on the compiler and OS? Can you explain? ConnectionState is an enum which is equivalent to an int, I think.

Reply:
On a 64-bits platform, the size of each field:
```
struct connection {
    ConnectionType *type;  ->8
    ConnectionState state;  ->4
    short int flags;  ->2
    short int refs;  ->2
    int last_errno;  ->4
    void *private_data;  ->8
    ConnectionCallbackFunc conn_handler;  ->8
    ConnectionCallbackFunc write_handler;  ->8
    ConnectionCallbackFunc read_handler;  ->8
    int fd; ->4
};
```
Originally, gdb shows
```
(gdb) p sizeof(connection)
$1 = 64
```

With the first patch in this PR:
gdb shows
```
(gdb) p sizeof(connection)
$1 = 56
```